ml4gw/transforms/injection.py

```diff
@@ -173,7 +174,9 @@ def __init__(
         # number of waveforms
         names = ["dec", "psi", "phi", "snr"]
         for name, param in zip(names, [dec, psi, phi, snr]):
-            if not isinstance(param, Callable) and param is not None:
+            if name == "snr" and isinstance(param, int):
+                self.register_buffer(name, param, persistent=False)
```

Comment:
You don't need to register it as a torch parameter, I don't think it will type check as an `int` in this case

Reply:
Ahhh, okay, I was getting my head turned around with all the types